plugins/services/src/js/events/SDKEndpointActions.js

```diff
@@ -0,0 +1,77 @@
+import { RequestUtil } from "mesosphere-shared-reactjs";
+
+import AppDispatcher from "#SRC/js/events/AppDispatcher";
+import {
+  REQUEST_SDK_ENDPOINTS_SUCCESS,
+  REQUEST_SDK_ENDPOINTS_ERROR,
+  REQUEST_SDK_ENDPOINT_SUCCESS,
+  REQUEST_SDK_ENDPOINT_ERROR,
+  REQUEST_SDK_ENDPOINTS_LOADING
+} from "../constants/ActionTypes";
+
+const SDKEndpointsActions = {
+  fetchEndpoints(serviceId) {
+    AppDispatcher.handleServerAction({
+      type: REQUEST_SDK_ENDPOINTS_LOADING,
+      data: { serviceId }
+    });
+    const url = `/service/${serviceId}/v1/endpoints`;
+    RequestUtil.json({
+      url,
+      method: "GET",
+      success(xhr) {
+        AppDispatcher.handleServerAction({
+          type: REQUEST_SDK_ENDPOINTS_SUCCESS,
+          data: { serviceId, endpoints: xhr }
+        });
+      },
+      error(xhr) {
+        AppDispatcher.handleServerAction({
+          type: REQUEST_SDK_ENDPOINTS_ERROR,
+          data: {
+            serviceId,
+            error: RequestUtil.parseResponseBody(xhr)
+          },
+          xhr
+        });
+      }
+    });
+  },
+
+  fetchEndpoint(serviceId, endpointName) {
+    const url = `/service/${serviceId}/v1/endpoints/${endpointName}`;
+    const fetch = new XMLHttpRequest();
```

Comment:
This is because we don't know the type of the response. the RequestUtil is only for JSON requests, so we used a generic fetch request since the response can be data, text, json or xml files.

Reply:
Okay, that should be already covered in CM 👍